File: SDKs/googleapis.com/dialogflow/v2/python/src/sdk/models/shared/googleclouddialogflowv2queryinput.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class GoogleCloudDialogflowV2QueryInput:
    r"""GoogleCloudDialogflowV2QueryInput
    Represents the query input. It can contain either: 1. An audio config which instructs the speech recognizer how to process the speech audio. 2. A conversational query in the form of text,. 3. An event that specifies which intent to trigger.
    """
    
    audio_config: Optional[GoogleCloudDialogflowV2InputAudioConfig] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('audioConfig') }})
    event: Optional[GoogleCloudDialogflowV2EventInput] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('event') }})
    text: Optional[GoogleCloudDialogflowV2TextInput] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('text') }})
    

File: SDKs/googleapis.com/dialogflow/v2/typescript/src/sdk/models/shared/googleclouddialogflowv2queryinput.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { GoogleCloudDialogflowV2InputAudioConfig } from "./googleclouddialogflowv2inputaudioconfig";
import { GoogleCloudDialogflowV2EventInput } from "./googleclouddialogflowv2eventinput";
import { GoogleCloudDialogflowV2TextInput } from "./googleclouddialogflowv2textinput";



// GoogleCloudDialogflowV2QueryInput
/** 
 * Represents the query input. It can contain either: 1. An audio config which instructs the speech recognizer how to process the speech audio. 2. A conversational query in the form of text,. 3. An event that specifies which intent to trigger.
**/
export class GoogleCloudDialogflowV2QueryInput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=audioConfig" })
  audioConfig?: GoogleCloudDialogflowV2InputAudioConfig;

  @SpeakeasyMetadata({ data: "json, name=event" })
  event?: GoogleCloudDialogflowV2EventInput;

  @SpeakeasyMetadata({ data: "json, name=text" })
  text?: GoogleCloudDialogflowV2TextInput;
}
